src/cli/main.c

```diff
@@ -33,19 +33,99 @@
 #include "common/image.h"
 #include "common/image_cache.h"
 #include "common/imageio.h"
+#include "common/imageio_jpeg.h"
 #include "common/imageio_module.h"
 #include "common/exif.h"
 #include "common/history.h"
+#include "control/conf.h"
+#include "develop/imageop.h"
 
 #include <sys/time.h>
 #include <unistd.h>
 #include <inttypes.h>
 #include <libintl.h>
 
+static void generate_thumbnail_cache()
+{
+  const int max_mip = DT_MIPMAP_2;
```

Comment:
well, i don't care too much. i'm using it as an int in the for() loop, so i thought the code is easier to understand this way.